plugins/c8y_configuration_plugin/Cargo.toml

```diff
@@ -17,13 +17,17 @@ assets = [
 ]
 
 [dependencies]
+agent_interface = { path = "../../crates/core/agent_interface" }
 anyhow = "1.0"
 c8y_api = { path = "../../crates/core/c8y_api" }
 c8y_smartrest = { path = "../../crates/core/c8y_smartrest" }
 clap = { version = "3.2", features = ["cargo", "derive"] }
 csv = "1.1"
 download = { path = "../../crates/common/download" }
+futures = "0.3"
+json = "0.12"
 mqtt_channel = { path = "../../crates/common/mqtt_channel" }
+reqwest = { version = "0.11", default-features = false, features = ["rustls-tls"] }
```

Comment:
Why is `reqwest` required?

From my understanding, the configuration plugin doesn't manage the HTTP interactions with the child-devices. This is one of the key benefits to *move* the file under the agent http root instead of having to put them over HTTP.

Reply:
after uploading a config snapshot, we make a http request to DELETE this file from /var/tedge. Should it be removed with tokio::fs instead?

Reply:
Yes, removing the file is better.

The point is to be symmetric.
* Either we move and remove the files directly under the agent HTTP root,
* Or we use HTTP put and delete, to do the same without direct access to the agent HTTP root.

The first option is by far easier to implement. So go for it. We might consider the second option later - if someone has a valid use case where the agent and the configuration plugin are not installed on the same host.

Reply:
Yes, I agree we should be consistent here. But can this wait for a follow-up PR?